src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -56,43 +59,56 @@ function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
     } else if (JsonRpcProvider.isProvider((provider as any).provider)) {
       throw new Error('Eip1193Bridge is experimental: pass your ethers Provider directly')
     } else {
-      ;[connector, hooks] = initializeConnector((actions) => new EIP1193(actions, provider))
+      ;[connector, hooks] = initializeConnector((actions) => new EIP1193({ actions, provider }))
     }
     connector.activate()
     return { connector, hooks }
   }
   return EMPTY_STATE
 }
 
+export let connections: [Connector, Web3ReactHooks][] = []
+
 interface ActiveWeb3ProviderProps {
   jsonRpcEndpoint?: string | JsonRpcProvider
   provider?: Eip1193Provider | JsonRpcProvider
 }
 
 export function ActiveWeb3Provider({
   jsonRpcEndpoint,
-  provider,
+  provider: propsProvider,
   children,
 }: PropsWithChildren<ActiveWeb3ProviderProps>) {
+  const metaMaskConnection = useMemo(
```

Comment:
if the connectors are completely memoized I would move them outside of the component, though I think we could switch to using the `Web3ReactProvider` as commented below